includes/abstracts/abstract-wc-product.php

```diff
@@ -1420,6 +1420,10 @@ public function is_visible() {
 			$visible = false;
 		}
 
+		if ( ( $parent_id = wp_get_post_parent_id( $this->get_id() ) ) && 'publish' !== get_post_status( $parent_id ) ) {
```

Comment:
This should use the product class' `get_parent_id` method.

Reply:
Same for the post status check. Use the methods. We should not assume products are posts in this class.